### PR TITLE
Allow GeoJSON to be serialized according to the right-hand rule

### PR DESCRIPTION
**Pull request by tschaub**
We don't enforce any coordinate ordering when constructing geometries.  For interoperability with other software, it should be possible to serialize geometries (polygons and multi-polygons in particular) following the right-hand rule (see for example elastic/elasticsearch#10227 for related discussion).

With this change, GeoJSON can be serialized in a predictable order.  By default, we serialize coordinates in the order they are constructed.  By passing the `rightHanded: true` write option, coordinates can be serialized following the right-hand rule (and `rightHanded: false` orients coordinates following the left-hand rule).

To support this, `ol.geom.Polygon#getCoordinates` is given an `opt_right` argument.  Same for `MultiPolygon`.

The one odd bit in these changes is that `rightHanded` is added to the `olx.format.ReadOptions` typedef.  This has no effect but is required to properly do type checking where `ol.format.Feature#adaptOptions` is used.  If we want to avoid this addition to the `ReadOptions` typedef, we could would split `adaptOptions` into `adaptReadOptions` and `adaptWriteOptions`.

### CODE REVIEW

File: externs/olx.js [hunk 1]
```diff
@@ -1594,6 +1596,22 @@ olx.format.WriteOptions.prototype.featureProjection;
...
+ * as they are provided at construction.  If `true`, the right-hand rule will
+ * be applied.  If `false`, the left-hand rule will be applied (clockwise for
+ * exterior and counter-clockwise for interior rings).  Note that not all
+ * formats support this.  The GeoJSON format does use this property when writing
+ * geometries.
+ *
+ * @type {boolean|undefined}
+ * @api
+ */
+olx.format.WriteOptions.prototype.rightHanded;
```
Comment (elemoine) [inline externs/olx.js]: Don't you want to add `olx.format.ReadOptions.prototype.rightHanded` as well?

Comment (elemoine) [inline externs/olx.js]: > Don't you want to add olx.format.ReadOptions.prototype.rightHanded as well?

Forget it, I now understand why you did not add it.

**Comment by elemoine** [comment]
LGTM. Please merge.

**Comment by tschaub** [comment]
Thanks for the review @elemoine.

**Pull request merged**